src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/TaskAwaiter.cs

```diff
@@ -209,7 +209,13 @@ internal static void OnCompletedInternal(Task task, Action continuation, bool co
 
             // If TaskWait* ETW events are enabled, trace a beginning event for this await
             // and set up an ending event to be traced when the asynchronous await completes.
-            if (TplEtwProvider.Log.IsEnabled() || Task.s_asyncDebuggingEnabled)
+            if (
+#if CORECLR
+                TplEtwProvider.Log.IsEnabled() || Task.s_asyncDebuggingEnabled
```

Comment:
As follow up step, it would be nice to do unification of `TplEtwProvider.Log.IsEnabled() || Task.s_asyncDebuggingEnabled` vs. `TaskTrace.Enabled` to reduce the diffs in this and other files.